argparse.sh

```diff
@@ -0,0 +1,189 @@
+#!/bin/bash
```

Comment:
This file adds a lot of LOC and I don't feel that it's tightly coupled. Can you share why you said it couples tightly to the launcher script?

With the launcher script adding a lot of LOC in this change, do you think it makes sense to go the opposite direction, and break out the launcher script into individual files? It's not yet at the point where it really needs that, but if it keeps growing that will eventually be a tech debt item and maybe it could be better to just get that out of the way now. Completely up to you though.

Reply:
`DESCRIPTION` and `COPYRIGHT` are just assumed to be set in the calling script, and the signature of the management script is not that common in other CLI tools. Thus, compared to tools like `getopt` or `getopts` this feels somewhat more tightly coupled.

I would much rather have one larger script than many small libraries stuck together (at least in bash). The scoping in bash is a mess and I find it easier to grasp larger bash scripts. I don't think we have hit the point yet that the script has become unwieldy.